src/lib/query/search.ts

```diff
@@ -78,14 +78,16 @@ export async function searchMemory(
   );
 
   // Check for any previous deep research results
-  let deepResearchResults: string | undefined;
+  let formattedResult = formatSearchResultsAsXml(rerankedResults);
   
   if (conversationId) {
     try {
       const deepResearch = await getDeepResearchResult(userId, conversationId);
       
       if (deepResearch) {
-        deepResearchResults = `<deep_research query="${deepResearch.query}">${deepResearch.formattedResult}</deep_research>`;
+        // Integrate deep research results directly into the main formatted results
+        formattedResult = formattedResult.replace("</search_results>", 
```

Comment:
what?? no no don't replace things. can't you insert the nodes / edges found through deep research into the ones already in the search so that the native formatAsXml just merges both and returns everything?